apps/communications/contacts/js/match_service.js

```diff
@@ -1,39 +1,48 @@
 'use strict';
-/* global utils */
 
 (function(exports) {
   var matchService = {};
 
   const CONTACTS_APP_ORIGIN = location.origin;
-
-  var extensionFrame = document.createElement('iframe');
-  extensionFrame.id = 'match-extension';
-  document.body.appendChild(extensionFrame);
   
+  var extensionFrame = null;
+
   var currentURI;
 
   matchService.match = function(contactId) {
-    extensionFrame.src = currentURI = 
-      '/contacts/views/matching/matching_contacts.html?contactId=' + contactId;
+    extensionFrame = window.open(
+      '/contacts/views/matching/matching_contacts.html?contactId=' +
+      contactId, 'Contacts match');
+    currentURI = '/contacts/views/matching/matching_contacts.html?contactId=' +
```

Comment:
make currentURI variable available before you do the window.open, and use that variable in the window.open